_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,201 @@
+---
+id: d0f69e
+name: All table header cells have corresponding data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has corresponding data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+authors:
+  - Jey Nandakumar
+  - Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any HTML element that is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of an element having a [semantic role](#semantic-role) of either [table](https://www.w3.org/TR/wai-aria-1.1/#table) or [grid](https://www.w3.org/TR/wai-aria-1.1/#grid), and is [visible](#visible) as well as [included in the accessibility tree](#included-in-the-accessibility-tree), where the element has any of the following [semantic roles](#semantic-role):
+
+- [cell](https://www.w3.org/TR/wai-aria-1.1/#cell)
```

Comment:
I am not sure if I fully understand the applicability. How does the rule handle data cells that have no descendants?

Reply:
@DagfinnRomen we are referring to descendants of any element with a role of `table` or `grid` which are the cells. Does that help?

Reply:
I am trying to wrap my head around it, and I may be going out on a tangent here. Following the DOM standard, an object A is called a descendant of an object B, if either A is a child of B or A is a child of an object C that is a descendant of B.

With the applicability as it is now, the rule applies to all cells in the table/grid. That is both rowheaders, coumnheaders and data cells. This is also mentioned in the listed semantic roles, that includes both cells and gridcells.

What will the outcome be when testing a "regular" data cell that has no descendants? It would not make sense that these cells even be tested as header cells.

Would it be better to not include cells and gridcells in the applicability?

Reply:
It's also unclear to me why you are applying this to elements with cell and gridcell semantic roles. Since you're targeting headers, aren't rowheader and columnheader enough?